src/main/java/user_use_cases/UserGateway.java

```diff
@@ -0,0 +1,254 @@
+package user_use_cases;
+
+import Interfaces.ReviewGatewayInterface;
+import entities.User;
+import entities.Review;
+import Gateways.ReviewGateway;
+
+import java.io.BufferedWriter;
+import java.util.*;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+public class UserGateway implements UserGatewayInterface{
+    /*
+    UserDatabase.csv -> How Instance Variables are saved
+     Column 0: username (String)
+     Column 1: password (String)
+     Column 2: past_reviews (ArrayList<Reviews>) - Use delimiter "/", to separate Review.id values.
+     Column 3: likedReviews (ArrayList<String>) - Use delimiter "|" to separate Review.id string values.
+     Column 4: received_reports (int)
+     Column 5: banned (Boolean) -> 0 - false, 1 - true
+     */
+
+    private static final String NAME_OF_USER_DATABASE = "src/main/java/Databases/UserDatabase.csv";
+
+    @Override
+    public void updateUser(User user) {
+
+        String new_username = user.getUsername();
+        String new_password = user.getPassword();
+
+        ArrayList<Review> past_review_list = user.getPast_reviews();
+        String past_reviews_string = "";
+        for (int i = 0; i < past_review_list.size() - 1; i++) {
+            past_reviews_string = past_reviews_string + past_review_list.get(i).getID() + "/";
```

Comment:
since what past_review_list now contains are arraylist of string ID, no need and cannot use .getID()